src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1175,7 +1176,8 @@ size_t QgsOgrProvider::layerCount() const
  */
 QgsWkbTypes::Type QgsOgrProvider::wkbType() const
 {
-  return static_cast<QgsWkbTypes::Type>( mOGRGeomType );
+  QgsWkbTypes::Type wkb = static_cast<QgsWkbTypes::Type>( mOGRGeomType );
+  return ogrDriverName == QLatin1String( "ESRI Shapefile" ) ? QgsWkbTypes::multiType( wkb ) : wkb;
```

Comment:
This is only strictly needed if wkb == LineString or Polygon. Shapefiles have a clear distinction beetween Point and MultiPoint

If we do this promotion of the reported type, I wonder if we should not also promote for consistency the geometries returned by the feature iterator to Multi too. I'm not sure which expectation the rest of the QGIS code base has regarding this, or if it is OK that a MultiPolygon layer has a mix of Polygon and MultiPolygon